pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -0,0 +1,266 @@
+          pinotQuery.setFilterExpression(toExpression(selectSqlNode.getWhere()));
+        }
+        if (selectSqlNode.getGroup() != null) {
+          pinotQuery.setGroupByList(convertSelectList(selectSqlNode.getGroup()));
+        }
+        break;
+      default:
+        throw new RuntimeException(
+            "Unable to convert SqlNode: " + sqlNode + " to PinotQuery. Unknown node type: " + sqlNode.getKind());
+    }
+    return pinotQuery;
+  }
+
+  private static List<String> extractOptionsFromSql(String sql) {
+    List<String> results = new ArrayList<>();
+    Matcher matcher = OPTIONS_REGEX_PATTEN.matcher(sql);
+    while (matcher.find()) {
+      results.add(matcher.group(1));
+    }
+    return results;
+  }
+
+  private static String removeOptionsFromSql(String sql) {
+    Matcher matcher = OPTIONS_REGEX_PATTEN.matcher(sql);
+    return matcher.replaceAll("");
+  }
+
+  private static List<Expression> convertSelectList(SqlNodeList selectList) {
+    List<Expression> selectExpr = new ArrayList<>();
+    final Iterator<SqlNode> iterator = selectList.iterator();
+    while (iterator.hasNext()) {
+      final SqlNode next = iterator.next();
+      selectExpr.add(toExpression(next));
+    }
+    return selectExpr;
+  }
+
+  private static List<Expression> convertOrderByList(SqlNodeList orderList) {
+    List<Expression> orderByExpr = new ArrayList<>();
+    final Iterator<SqlNode> iterator = orderList.iterator();
+    while (iterator.hasNext()) {
+      final SqlNode next = iterator.next();
+      orderByExpr.add(convertOrderBy(next));
+    }
+    return orderByExpr;
+  }
+
+  private static Expression convertOrderBy(SqlNode node) {
+    final SqlKind kind = node.getKind();
+    Expression expression;
+    switch (kind) {
+      case IDENTIFIER:
+        expression = RequestUtils.getFunctionExpression("ASC");
+        expression.getFunctionCall().addToOperands(toExpression(node));
+        break;
+      case DESCENDING:
+        SqlBasicCall basicCall = (SqlBasicCall) node;
+        expression = RequestUtils.getFunctionExpression("DESC");
+        expression.getFunctionCall().addToOperands(toExpression(basicCall.getOperands()[0]));
+        break;
+      default:
+        throw new RuntimeException("Unknown node type: " + node.getKind());
+    }
+    return expression;
+  }
+
+  private static Expression toExpression(SqlNode node) {
+    // System.out.printf("Current processing SqlNode: %s, node.getKind(): %s\n", node, node.getKind());
+    LOGGER.debug("Current processing SqlNode: {}, node.getKind(): {}", node, node.getKind());
+    switch (node.getKind()) {
+      case IDENTIFIER:
+        if (((SqlIdentifier) node).isStar()) {
+          return RequestUtils.getIdentifierExpression("*");
+        }
+        if (((SqlIdentifier) node).isSimple()) {
+          return RequestUtils.getIdentifierExpression(((SqlIdentifier) node).getSimple());
+        }
+        return RequestUtils.getIdentifierExpression(node.toString());
```

Comment:
Seems like this part takes care of expressions (eg `select foo(a, bar(c, d)...`)?

Reply:
yes, it will convert expressions inside select/filtering/groupby  recursively.